integration-libs/epd-visualization/components/visual-picking/visual-picking-tab/visual-picking-tab.service.ts

```diff
@@ -37,26 +39,42 @@ export class VisualPickingTabService {
     this.visualViewerService = visualViewerService;
     this.visualPickingProductListService = visualPickingProductListService;
 
-    this.visualViewerService.visualizationLoadInfoChange.subscribe(
-      this.handleLoadVisualizationInfoChange.bind(this)
-    );
+    this.visualizationLoadInfoChangeSubscription =
+      this.visualViewerService.visualizationLoadInfoChange.subscribe(
+        this.handleLoadVisualizationInfoChange.bind(this)
+      );
+
+    this.getProductReferences$Subscription =
+      this.visualPickingProductListService
+        .getProductReferences$()
+        .subscribe((productReferences: ProductReference[]) => {
+          this.setProductReferences(productReferences);
+          if (productReferences.length > 0) {
+            this.visualPickingProductListService.currentProduct$
+              .pipe(first())
+              .subscribe((currentProduct: Product) => {
+                this.visualViewerService
+                  .loadVisualization(currentProduct.code as string)
+                  .pipe(first())
+                  .subscribe();
+              });
+          }
+        });
+  }
 
-    this.visualPickingProductListService
-      .getProductReferences$()
-      .subscribe((productReferences: ProductReference[]) => {
-        this.setProductReferences(productReferences);
-        if (productReferences.length > 0) {
-          this.visualPickingProductListService.currentProduct$.subscribe(
-            (currentProduct: Product) => {
-              this.visualViewerService
-                .loadVisualization(currentProduct.code as string)
-                .subscribe();
-            }
-          );
-        }
-      });
+  ngOnDestroy(): void {
+    this.visualizationLoadInfoChangeSubscription.unsubscribe();
+    this.getProductReferences$Subscription.unsubscribe();
+
+    if (this.getFilteredProductReferences$Subscription) {
+      this.getFilteredProductReferences$Subscription.unsubscribe();
+    }
```

Comment:
hint: using _optional chaining_ you can avoid `if` in this case and shorten it to `this.getFilteredProductReferences$Subscription?.unsubscribe();` . No need to change. Just FYI

Reply:
I have used optional chaining in these places.